spec/integration/category.spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe 'Categories', type: :feature do
+  describe 'index' do
+    before(:each) do
+      @user = User.create(name: 'tushar', email: 'tushar@mail.com', password: 'budget')
+      sign_in @user
+      @category = Category.create(author: @user, name: 'Food', icon: 'https://cdn-icons-png.flaticon.com/512/2454/2454273.png')
+      @total = @category.payments.sum(:amount)
+      visit categories_path
+    end
+
+    it 'renders name of category' do
+      expect(page).to have_content(@category.name)
+    end
+
+    it 'renders icon of the category' do
+      expect(page).to have_css("img[src*='https://cdn-icons-png.flaticon.com/512/2454/2454273.png']")
+    end
+
+    it 'renders total transaction of the category' do
+      expect(page).to have_content(@total)
+    end
+
+    it 'renders remove button' do
+      expect(page).to have_content('remove')
+    end
+
+    it 'redirects to delete path' do
+      click_button 'remove'
+      expect(page).to have_current_path(categories_path)
+    end
+
+    it 'renders a button to add category' do
+      expect(page).to have_content('Add Category')
+    end
+
+    it 'redirects to a form for new category' do
+      click_link 'Add Category'
+      expect(page).to have_current_path(new_category_path)
+    end
+  end
+end
```

Comment:
- [ ] Please fix your test cases
<img width="1097" alt="Screen Shot 2022-11-17 at 20 25 26" src="https://user-images.githubusercontent.com/44745855/202528150-578d9e78-247d-45b6-b200-ba7b0f11456a.png">

Reply:
Hi @DeliceLydia , Again these are not my integration or models tests, these are default generated by rails, still I have removed these files so it will be easier to see the test cases, please use the command `rspec spec/integration/*spec.rb` for integration testing and `rspec spec/models/*spec.rb` for models tests, the general command is `rspec spec/directory_name/*_spec.rb`
see below :point_down: 
![image](https://user-images.githubusercontent.com/97521557/202531796-e41e3681-3988-45b3-bed1-fbdb0273f2a1.png)